src/cogent3/core/new_genetic_code.py

```diff
@@ -127,9 +127,6 @@ def __post_init__(self, ncbi_code_sequence: str, ncbi_start_codon_map: str):
     def __hash__(self):
         return hash(self.name)
 
-    def __eq__(self, other):
```

Comment:
Is the duplicate `__eq__` defined on L162 the correct implementation?

Reply:
I think this one is correct. Please delete the other -- good catch!

Reply:
Deleting the other throws:
```
.../test_new_sequence.py: 
    from cogent3.core import (
.../new_genetic_code.py:407:
    _CODES[code] = code
.../new_genetic_code.py:132: in __eq__
    return self.name == other.name
E   AttributeError: 'str' object has no attribute 'name'
```

Reply:
then change code to
```python
return self.name == getattr(other, "name", None)
```